From 75293d6a8835d63a280eda6283d001eac8edd165 Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Wed, 19 Oct 2022 10:22:39 +0200
Subject: [PATCH] Recognize SystemSettingsExternalApp again

Many .desktop files still use X-KDE-ServiceTypes=SystemSettingsExternalApp,
there's no good reason to drop support for that. Just look at the ServiceType
again.


(cherry picked from commit 7fe2abd92285d0fa3c0082c6563feb3d17169abe)
---
 core/MenuItem.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/MenuItem.cpp b/core/MenuItem.cpp
index b1da9f18..e288efbe 100644
--- a/core/MenuItem.cpp
+++ b/core/MenuItem.cpp
@@ -191,7 +191,7 @@ void MenuItem::setMetaData(const KPluginMetaData &data)
     d->comment = data.description();
     d->iconName = data.iconName();
     d->systemsettingsCategoryModule = data.value(QStringLiteral("X-KDE-System-Settings-Category-Module"));
-    d->isExternalAppModule = data.value(QStringLiteral("IsExternalApp"), false);
+    d->isExternalAppModule = data.value(QStringLiteral("IsExternalApp"), false) || data.serviceTypes().contains(QStringLiteral("SystemSettingsExternalApp"));
 }
 
 KPluginMetaData MenuItem::metaData()
-- 
2.38.0

